core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -537,6 +559,22 @@ public void onClosed(Channel c, IOException cause) {
         Jenkins.getInstance().getQueue().scheduleMaintenance();
     }
 
+    /*package*/ static boolean isRelativePath(String path) {
+        if (path.startsWith("/"))
+            return false;
```

Comment:
`\windows\temp\slave` is relative, `/windows/temp/slave` is fine to consider as absolute as you are indicating that you think it is unix and the `isUnix` test will warn you about the path being for a different OS